app/CMakeLists.txt

```
@@ -95,7 +95,10 @@ target_sources_ifdef(CONFIG_ZMK_BATTERY_REPORTING app PRIVATE src/battery.c)
 
 target_sources_ifdef(CONFIG_ZMK_HID_INDICATORS app PRIVATE src/events/hid_indicators_changed.c)
 
-target_sources_ifdef(CONFIG_ZMK_SPLIT app PRIVATE src/events/split_peripheral_status_changed.c)
+if (CONFIG_ZMK_SPLIT AND (NOT CONFIG_ZMK_SPLIT_ROLE_CENTRAL))
+  target_sources(app PRIVATE src/events/split_peripheral_status_changed.c)
+endif()
+target_sources_ifdef(CONFIG_ZMK_SPLIT_ROLE_CENTRAL app PRIVATE src/events/split_central_peripheral_status_changed.c)
```

Honestly, it'd be clearer if using just conditional branches for both, IMHO:

```suggestion
if (CONFIG_ZMK_SPLIT)
  if(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
    target_sources(app PRIVATE src/events/split_central_peripheral_status_changed.c)
  else()
    target_sources(app PRIVATE src/events/split_peripheral_status_changed.c)
  endif()
endif()
```

---

Seems addressed, resolving.